core/src/main/java/io/aiven/kafka/tieredstorage/manifest/SegmentIndexesV1Builder.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2023 Aiven Oy
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.aiven.kafka.tieredstorage.manifest;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import io.aiven.kafka.tieredstorage.storage.BytesRange;
+
+import static org.apache.kafka.server.log.remote.storage.RemoteStorageManager.IndexType;
+
+public class SegmentIndexesV1Builder {
+    final Map<IndexType, SegmentIndexV1> indexes = new HashMap<>(IndexType.values().length);
+    int currentPosition = 0;
+
+    public SegmentIndexesV1Builder add(final IndexType indexType, final BytesRange range) {
```

Comment:
It seems from the range we only need the size. Maybe we should pass it instead? This will also eliminate the problem of the need to check for overlapping ranges, extra tests for this, etc.